### PR TITLE
Use Stride Kinds/IDs in Unstructured Nabla

### PR DESCRIPTION
**Pull request by fthaler**
Slightly reduces memory usage required for strides / kernel arguments. Gives speedups of almost 9% on Clang/Cray CUDA and around 3.5% on NVCC (for fn_unstructured_nabla_fused_tuple_of_fields, double precision, large domain).

### CODE REVIEW

**Comment by fthaler** [comment]
launch perftest

**Comment by fthaler** [comment]
launch jenkins

**Comment by fthaler** [comment]
launch perftest

**Review by havogt — APPROVED** [review]

**Comment by fthaler** [comment]
launch jenkins

**Comment by havogt** [comment]
Can you summarize in the commit message how performance changes.
